python-package/lightgbm/sklearn.py

```diff
@@ -285,6 +285,8 @@ def get_params(self, deep=True):
     def set_params(self, **params):
         for key, value in params.items():
             setattr(self, key, value)
+            if hasattr(self, '_' + key):
```

Comment:
if an params was named `other_params`, could that disable line 290?

Reply:
@chivee Yeah, you're right!
Any parameter with underscore (`_Booster`, `_classes`, especially `_n_features`) will break normal logic in some way. I think that we're assuming that LightGBM user is rather smart person and wont do this 😃 .